Support Azure Workload Identity (#1048)
fluxcd/source-controller: docs/spec/v1beta2/buckets.md

        aadpodidbinding: ${IDENTITY_NAME}  # match the AzureIdentity name
```

If you have set aad-pod-identity up correctly and labeled the source-controller
pod, then you don't need to reference a secret.

```yaml
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: Bucket
metadata:
  name: azure-bucket
  namespace: flux-system
spec:
  interval: 5m0s
  provider: azure
  bucketName: testsas
  endpoint: https://testfluxsas.blob.core.windows.net
```

> ##### Workload Identity

Reviewer: Given the other option is deprecated, I would move this above the other one.

> If you have [workload identity mutating webhook](https://azure.github.io/azure-workload-identity/docs/installation/managed-clusters.html)

Reviewer: ```suggestion
If you have a [Workload Identity mutating webhook](https://azure.github.io/azure-workload-identity/docs/installation/managed-clusters.html)
```

> installed on your cluster. You would need to create an azure identity and

Reviewer: ```suggestion
installed on your cluster. You would need to create an Azure Identity and
```

give it access to Azure Blob Storage.

```shell
export IDENTITY_NAME="blob-access"

az role assignment create --role "Storage Blob Data Reader" \
--assignee-object-id "$(az identity show -n $IDENTITY_NAME  -o tsv --query principalId  -g $RESOURCE_GROUP)" \
--scope "/subscriptions/<SUBSCRIPTION-ID>/resourceGroups/<RESOURCE_GROUP>/providers/Microsoft.Storage/storageAccounts/<account-name>/blobServices/default/containers/<container-name>"
```

Establish federated identity between the identity and the source-controller
serviceaccount.

```shell
export SERVICE_ACCOUNT_ISSUER="$(az aks show --resource-group <RESOURCE_GROUP> --name <CLUSTER-NAME> --query "oidcIssuerProfile.issuerUrl" -otsv)"

az identity federated-credential create \
  --name "kubernetes-federated-credential" \
  --identity-name "${IDENTITY_NAME}" \
  --resource-group "${RESOURCE_GROUP}" \
  --issuer "${SERVICE_ACCOUNT_ISSUER}" \
  --subject "system:serviceaccount:flux-system:source-controller"
```

> Add a patch to label and annotate the source-controller pods and serviceaccount

Reviewer: ```suggestion
Add a patch to label and annotate the source-controller Pods and ServiceAccount
```

correctly so that it can match an identity binding:

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
  - gotk-components.yaml
  - gotk-sync.yaml
patches:
  - patch: |-
      apiVersion: v1
      kind: ServiceAccount
      metadata:
        name: source-controller
        namespace: flux-system
        annotations:
          azure.workload.identity/client-id: <AZURE_CLIENT_ID>
        labels:
          azure.workload.identity/use: "true"
  - patch: |-
      apiVersion: apps/v1
      kind: Deployment
      metadata:
        name: source-controller
        namespace: flux-system
        labels:
          azure.workload.identity/use: "true"
      spec:
        template:
          metadata:
            labels:
              azure.workload.identity/use: "true"
```

> If you have set workload identity up correctly and labeled the source-controller
> pod, then you don't need to reference a secret. For more information,
> please see [documentation](https://azure.github.io/azure-workload-identity/docs/quick-start.html)

Reviewer: ```suggestion
If you have set up Workload Identity correctly and labeled the source-controller Pod,
then you don't need to reference a Secret. For more information, refer to the [Azure
documentation](https://azure.github.io/azure-workload-identity/docs/quick-start.html).
```
